app/controllers/api/pxe_servers_controller.rb

```diff
@@ -1,5 +1,91 @@
 module Api
   class PxeServersController < BaseController
+    INVALID_PXE_SERVER_ATTRS = %w[id href].freeze # Cannot update or create these
+    REQUIRED_PXE_SERVER_ATTRS = %w[name uri].freeze
+    REQUIRED_PXE_MENU_ATTRS = %w[file_name].freeze
+    REQUIRED_PXE_AUTH_ATTRS = %w[userid password].freeze
+
     include Subcollections::PxeImages
+    include Subcollections::PxeMenus
+
+    def create_resource(_type, _id, data = {})
+      PxeServer.transaction do
+        authentication = data.delete('authentication')
+        menus = data.delete('pxe_menus')
+
+        validate_pxe_server_data(data)
+        validate_pxe_create_data(data, REQUIRED_PXE_SERVER_ATTRS, 'pxe server')
+
+        server = collection_class(:pxe_servers).new(data)
+        # generates uir_prefix which is ched if server needs authentication or not
```

Comment:
this is good comment 👍 (it is not obvious from `verify_uri_prefix_before_save` why we need it)